tests/test_iostream.cpp

```diff
@@ -25,6 +27,40 @@ void noisy_funct_dual(std::string msg, std::string emsg) {
     std::cerr << emsg;
 }
 
+// object to manage C++ thread
+// simply repeatedly write to std::cerr until stopped
+// redirect is called at some point to test the safety of scoped_estream_redirect
+struct TestThread {
+    TestThread() : t_{nullptr}, stop_{false} {
+        auto thread_f = [this] {
+            while(!this->stop_) {
```

Comment:
Seems the formatting CI is still complaining about this `while` not having a space before the `(`.

Reply:
I'm fine fixing these last few things - the include sequence, remove the comment, and remove the extra new lines